packages/next/src/client/app-index.tsx

```diff
@@ -172,7 +173,8 @@ const initialServerResponse = createFromReadableStream(readable, {
 })
 
 function ServerRoot(): React.ReactNode {
-  const initialRSCPayload = use<InitialRSCPayload>(initialServerResponse)
+  const initialResponse = use<RSCPayload>(initialServerResponse)
+  const initialRSCPayload = getInitialRSCPayload(initialResponse)
```

Comment:
`use<RSCPayload>(initialServerResponse)` is already a type cast. Which is fine in this case, because we also control the server response. Therefore, narrowing it down to `InitialRSCPayload` with a runtime check that might throw an invariant error might be unnecessary. I think we could just as well declare it as `use<InitialRSCPayload>(initialServerResponse)`.

Reply:
Similar reasoning here to [this](https://github.com/vercel/next.js/pull/67680#discussion_r1679864486) since we don't fully control the response.